src/aliceVision/hdr/brackets.cpp

```diff
@@ -101,18 +103,62 @@ bool estimateBracketsFromSfmData(std::vector<std::vector<std::shared_ptr<sfmData
     return true;
 }
 
-void selectTargetViews(std::vector<std::shared_ptr<sfmData::View>> & out_targetViews, const std::vector<std::vector<std::shared_ptr<sfmData::View>>> & groups, int offsetRefBracketIndex)
+void selectTargetViews(std::vector<std::shared_ptr<sfmData::View>> & out_targetViews, const std::vector<std::vector<std::shared_ptr<sfmData::View>>> & groups, int offsetRefBracketIndex, const std::string& targetIndexesFilename)
 {
-    for(auto & group : groups)
+    // If targetIndexesFilename cannot be opened or is not valid target views are derived from the middle exposed views
+    // For odd number, there is no ambiguity on the middle image.
+    // For even number, we arbitrarily choose the more exposed view (as we usually have more under-exposed images than over-exposed).
+    const int viewNumberPerGroup = groups[0].size();
+    const int middleIndex = viewNumberPerGroup / 2;
+    const int targetIndex = middleIndex + offsetRefBracketIndex;
+
+    if ((targetIndex >= 0) && (targetIndex < viewNumberPerGroup))
     {
-        // Target views are the middle exposed views
-        // For add number, there is no ambiguity on the middle image.
-        // For even number, we arbitrarily choose the more exposed view (as we usually have more under-exposed images than over-exposed).
-        const int middleIndex = int(group.size()) / 2;
-        const int targetIndex = clamp(middleIndex + offsetRefBracketIndex, 0, int(group.size()) - 1);
+        ALICEVISION_LOG_INFO("Use offsetRefBracketIndex parameter");
+        for(auto & group : groups)
+        {
+            out_targetViews.push_back(group[targetIndex]);
+        }
+    }
+    else // try to use indexes in the file
+    {
+        ALICEVISION_LOG_INFO("offsetRefBracketIndex parameter out of range, read file containing target indexes");
+        std::vector<int> targetIndexes;
+        std::ifstream file(targetIndexesFilename);
+        if (!file)
+        {
+            ALICEVISION_LOG_WARNING("Unable to open the file " << targetIndexesFilename << " with the selection of exposures. This file would be needed to select the optimal exposure for the creation of the HDR images. Use clamped offsetRefBracketIndex parameter.");
```

Comment:
This should generate an error. We don't want a behavior change if there is a file system problem.